src/web/web_actions.nit

```diff
@@ -0,0 +1,163 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Nitcorn actions used by the nitweb server.
+module web_actions
+
+import web_views
+import model::model_collect
+
+# Display the tree of all loaded mentities.
+class TreeAction
+	super NitAction
+
+	# Model to explore and render.
+	var model: Model
+
+	# View to render.
+	var view = new HtmlHomePage(model) is lazy
+
+	redef fun answer(request, url) do return render_view(view)
+end
+
+# Display the list of mentities matching `namespace`.
+class SearchAction
+	super NitAction
+
+	# Model to explore and render.
+	var model: Model
+
+	# TODO handle more than full namespaces.
+	redef fun answer(request, url) do
+		var namespace = request.param("namespace")
+		if namespace == null or namespace.is_empty then
+			return render_error(400, "Missing :namespace.")
+		end
+		var mentities = model.collect_by_namespace(namespace)
+		if request.bool_arg("json") or else false then
```

Comment:
Since you might have a lot of cases where you check if `json` is asked, I suggest that you factorise the test. In the request class for instance.